app/drivers/nomad/linux_virtualizer.go

```diff
@@ -290,7 +290,11 @@ func (lv *LinuxVirtualizer) GetDestroyScriptGenerator() func(string, string) str
 		if [ $? -ne 0 ]; then
 		  %s stop -f %s; %s rm -f %s
 		fi
-	`, ignitePath, vm, ignitePath, vm, ignitePath, vm, ignitePath, vm)
+		vm_id=$(ignite ps | grep %s | awk '{print $1}')
```

Comment:
Is that the only way to handle it? What about the case you mentioned where line 291 goes out of Nomad execution?

Reply:
I was wrong about it, it goes not next line even if it fails. So for now forcefully removing firecracker vm from disk.

Reply:
do we have a plan to figure this out, again i assume we get cni error, do you think this is the case?
![image](https://github.com/user-attachments/assets/6e833522-008a-47f4-ba8c-14263a911d1f)

Reply:
Removing the firecracker VM all together should help with this no ?